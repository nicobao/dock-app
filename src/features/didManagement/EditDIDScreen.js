import React, {useEffect} from 'react';
import {
  BackButton,
  Box,
  Header,
  Input,
  LoadingButton,
  NBox,
  ScreenContainer,
  Typography,
} from '../../design-system';
import {navigateBack} from '../../core/navigation';
import {translate} from '../../locales';
import {FormControl, ScrollView, Stack} from 'native-base';
import {addTestId} from '../../core/automation-utils';
import {useDIDManagementHandlers} from './didHooks';
export function EditDIDScreen({handleChange, form, handleSubmit}) {
  return (
    <ScreenContainer {...addTestId('EditDIDScreen')}>
      <Header>
        <Box
          marginLeft={1}
          marginRight={22}
          flexDirection="row"
          alignItems="center">
          <NBox width={'80px'}>
            <BackButton
              {...addTestId('EditDIDScreenGoBack')}
              onPress={navigateBack}
            />
          </NBox>
          <NBox
            flex={1}
            width="100%"
            alignContent="center"
            alignItems="center"
            pl={15}>
            <Typography variant="h3">
              {translate('didManagement.edit_did')}
            </Typography>
          </NBox>
          <NBox width="80px" alignItems="flex-end" />
        </Box>
      </Header>
      <ScrollView marginLeft={5} marginRight={5}>
        <FormControl>
          <Stack mt={7}>
            <FormControl.Label>
              {translate('didManagement.did_name')}
            </FormControl.Label>
            <Input
              {...addTestId('EditDIDScreenDIDName')}
              value={form.didName}
              onChangeText={handleChange('didName')}
              autoCapitalize="none"
            />
          </Stack>
        </FormControl>
      </ScrollView>
      <NBox mx={7}>
        <LoadingButton
          {...addTestId('EditDIDScreenSave')}
          full
          testID="save-btn"
          mb={70}
          onPress={handleSubmit}
          isDisabled={form.didName.trim().length <= 0}>
          {translate('didManagement.save')}
        </LoadingButton>
      </NBox>
    </ScreenContainer>
  );
}

export function EditDIDScreenContainer({route}) {
  const {didDocumentResolution} = route.params;
  const {form, onEditDID, handleChange} = useDIDManagementHandlers();

  useEffect(() => {
    if (didDocumentResolution) {
      handleChange('id')(didDocumentResolution.id);
      handleChange('didName')(
        typeof didDocumentResolution.name === 'string'
          ? didDocumentResolution.name.trim()
          : '',
      );
    }
    //eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <EditDIDScreen
      handleChange={handleChange}
      form={form}
      handleSubmit={onEditDID}
    />
  );
}
