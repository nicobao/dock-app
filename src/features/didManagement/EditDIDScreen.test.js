import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {EditDIDScreen} from './EditDIDScreen';

const mockStore = configureMockStore();

describe('EditDIDScreen', () => {
  it('should render correctly', () => {
    const initialState = {
      // placeholder for redux store
    };

    const form = {
      didName: '',
      didType: 'diddock',
      showDIDDockQuickInfo: true,
      keypairType: 'ed25519',
      derivationPath: '',
      _errors: {
        didType: '',
      },
      _hasError: false,
    };

    const wrapper = shallow(
      <EditDIDScreen
        handleChange={jest.fn()}
        form={form}
        handleSubmit={jest.fn()}
      />,
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
  it('should render correctly when button is not enable', () => {
    const initialState = {
      // placeholder for redux store
    };

    const form = {
      didName: 'Name',
      didType: 'diddock',
      showDIDDockQuickInfo: true,
      keypairType: 'ed25519',
      derivationPath: '',
      _errors: {
        didType: '',
      },
      _hasError: false,
    };

    const wrapper = shallow(
      <EditDIDScreen
        handleChange={jest.fn()}
        form={form}
        handleSubmit={jest.fn()}
      />,
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
