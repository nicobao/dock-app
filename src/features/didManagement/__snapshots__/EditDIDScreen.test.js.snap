// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditDIDScreen should render correctly 1`] = `
<RNCSafeAreaView
  accessibilityLabel="EditDIDScreen"
  style={
    Object {
      "backgroundColor": "#141417",
      "flex": 1,
    }
  }
  testID="EditDIDScreen"
>
  <AppGlobalHeader />
  <Memo(ForwardRef(Box))
    flex={1}
  >
    <Memo(ForwardRef(Box))
      flex={1}
    >
      <Styled(View)>
        <Box
          alignItems="center"
          flexDirection="row"
          marginLeft={1}
          marginRight={22}
        >
          <Memo(ForwardRef(Box))
            width="80px"
          >
            <BackButton
              accessibilityLabel="EditDIDScreenGoBack"
              onPress={[MockFunction]}
              testID="EditDIDScreenGoBack"
            />
          </Memo(ForwardRef(Box))>
          <Memo(ForwardRef(Box))
            alignContent="center"
            alignItems="center"
            flex={1}
            pl={15}
            width="100%"
          >
            <Styled(Component)
              variant="h3"
            >
              Edit DID
            </Styled(Component)>
          </Memo(ForwardRef(Box))>
          <Memo(ForwardRef(Box))
            alignItems="flex-end"
            width="80px"
          />
        </Box>
      </Styled(View)>
      <ForwardRef
        marginLeft={5}
        marginRight={5}
      >
        <Memo(ForwardRef(FormControl))>
          <Memo(ForwardRef(Stack))
            mt={7}
          >
            <Memo(ForwardRef(FormControlLabel))>
              DID Name
            </Memo(ForwardRef(FormControlLabel))>
            <Memo(ForwardRef(Input))
              accessibilityLabel="EditDIDScreenDIDName"
              autoCapitalize="none"
              testID="EditDIDScreenDIDName"
              value=""
            />
          </Memo(ForwardRef(Stack))>
        </Memo(ForwardRef(FormControl))>
      </ForwardRef>
      <Memo(ForwardRef(Box))
        mx={7}
      >
        <LoadingButton
          accessibilityLabel="EditDIDScreenSave"
          full={true}
          isDisabled={true}
          mb={70}
          onPress={[MockFunction]}
          testID="save-btn"
        >
          Save
        </LoadingButton>
      </Memo(ForwardRef(Box))>
    </Memo(ForwardRef(Box))>
  </Memo(ForwardRef(Box))>
</RNCSafeAreaView>
`;

exports[`EditDIDScreen should render correctly when button is not enable 1`] = `
<RNCSafeAreaView
  accessibilityLabel="EditDIDScreen"
  style={
    Object {
      "backgroundColor": "#141417",
      "flex": 1,
    }
  }
  testID="EditDIDScreen"
>
  <AppGlobalHeader />
  <Memo(ForwardRef(Box))
    flex={1}
  >
    <Memo(ForwardRef(Box))
      flex={1}
    >
      <Styled(View)>
        <Box
          alignItems="center"
          flexDirection="row"
          marginLeft={1}
          marginRight={22}
        >
          <Memo(ForwardRef(Box))
            width="80px"
          >
            <BackButton
              accessibilityLabel="EditDIDScreenGoBack"
              onPress={[MockFunction]}
              testID="EditDIDScreenGoBack"
            />
          </Memo(ForwardRef(Box))>
          <Memo(ForwardRef(Box))
            alignContent="center"
            alignItems="center"
            flex={1}
            pl={15}
            width="100%"
          >
            <Styled(Component)
              variant="h3"
            >
              Edit DID
            </Styled(Component)>
          </Memo(ForwardRef(Box))>
          <Memo(ForwardRef(Box))
            alignItems="flex-end"
            width="80px"
          />
        </Box>
      </Styled(View)>
      <ForwardRef
        marginLeft={5}
        marginRight={5}
      >
        <Memo(ForwardRef(FormControl))>
          <Memo(ForwardRef(Stack))
            mt={7}
          >
            <Memo(ForwardRef(FormControlLabel))>
              DID Name
            </Memo(ForwardRef(FormControlLabel))>
            <Memo(ForwardRef(Input))
              accessibilityLabel="EditDIDScreenDIDName"
              autoCapitalize="none"
              testID="EditDIDScreenDIDName"
              value="Name"
            />
          </Memo(ForwardRef(Stack))>
        </Memo(ForwardRef(FormControl))>
      </ForwardRef>
      <Memo(ForwardRef(Box))
        mx={7}
      >
        <LoadingButton
          accessibilityLabel="EditDIDScreenSave"
          full={true}
          isDisabled={false}
          mb={70}
          onPress={[MockFunction]}
          testID="save-btn"
        >
          Save
        </LoadingButton>
      </Memo(ForwardRef(Box))>
    </Memo(ForwardRef(Box))>
  </Memo(ForwardRef(Box))>
</RNCSafeAreaView>
`;
